pkg/schemacmp/table_test.go

```diff
@@ -0,0 +1,410 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemacmp_test
+
+import (
+	"testing"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/ddl"
+	"github.com/pingcap/tidb/util/mock"
+	// initialize expression.EvalAsAst
+	_ "github.com/pingcap/tidb/planner"
+
+	. "github.com/pingcap/tidb-tools/pkg/schemacmp"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type tableSchema struct {
+	parser *parser.Parser
+	sctx   *mock.Context
+}
+
+var _ = Suite(&tableSchema{})
+
+func (s *tableSchema) SetUpSuite(c *C) {
+	s.parser = parser.New()
+	s.sctx = mock.NewContext()
+}
+
+func (s *tableSchema) toTableInfo(createTableStmt string) (*model.TableInfo, error) {
+	node, err := s.parser.ParseOneStmt(createTableStmt, "", "")
+	if err != nil {
+		return nil, err
+	}
+	createStmtNode, ok := node.(*ast.CreateTableStmt)
+	if !ok {
+		return nil, errors.New("not a create table statement")
+	}
+	return ddl.MockTableInfo(s.sctx, createStmtNode, 1)
+}
+
+func (s *tableSchema) TestJoinSchemas(c *C) {
+	testCases := []struct {
+		name    string
+		a       string
+		b       string
+		cmp     int
+		cmpErr  string
+		join    string
+		joinErr string
+	}{
+		{
+			name: "DM_002/1",
+			a:    "CREATE TABLE tb1 (col1 INT)",
+			b:    "CREATE TABLE tb2 (col1 INT, new_col1 INT)",
```

Comment:
do we need to add test case like
```
a:    "CREATE TABLE tb1 (col1 INT)",
b:    "CREATE TABLE tb2 (new_col1INT, col1 INT)",
```

Reply:
OK, though since we're using a map which is unordered, it doesn't really matter 😄